src/Shared/runtime/Http2/Hpack/HPackDecoder.cs

```diff
@@ -543,7 +543,12 @@ private void OnIndexedHeaderField(int index, IHttpHeadersHandler handler)
             else
             {
                 ref readonly HeaderField header = ref GetDynamicHeader(index);
+#if KESTREL
+                // Kestrel can skip some validation if it knows the header is indexed.
+                handler.OnDynamicIndexedHeader(header.StaticTableIndex, header.Name, header.Value);
+#else
                 handler.OnHeader(header.Name, header.Value);
+#endif
```

Comment:
I would really like to avoid these #ifs, they make the testing much harder. Why not call the new method for HttpClient too? Doesn't the DIM call the old one anyways?

Reply:
Yes. I'm not sure how runtime feels about DIMs which is why I left it specific to aspnetcore.

I'll remove the ifdefs and if anyone in runtime brings it up then we can re-evaluate.